File: leetcode/python/0083_remove_duplicates_from_sorted_list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if head == None: return None
        if head.next == None: return head

        res, curr, prev = head, head.next, head
        # while curr != None:
        # -> if curr.val != prev.val: prev.next = curr, prev = prev.next
        # curr = curr.next
        # return res
        while curr != None:
            if curr.val != prev.val:
                prev.next = curr
                prev = prev.next

            if prev.val == curr.val and curr.next == None:
                prev.next = None
            
            curr = curr.next

        return res

File: leetcode/java/0083_remove_duplicates_from_sorted_list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if (head == null) return null;
        if (head.next == null) return head;

        ListNode res = head, prev = head, curr = head.next;

        while (curr != null) {
            if (prev.val != curr.val) {
                prev.next = curr;
                prev = prev.next;
            }
            curr = curr.next;

            if (curr == null) prev.next = null;
        }

        return res;
    }
}